gstools/variogram/estimator.pyx

```diff
@@ -36,6 +36,27 @@ cdef inline double _distance_2d(
 ) nogil:
     return sqrt((x[i] - x[j]) * (x[i] - x[j]) + (y[i] - y[j]) * (y[i] - y[j]))
 
+cdef inline double _distance_2d_haversine(
+    const double[:] x,
+    const double[:] y,
+    const double[:] z,
+    const int i,
+    const int j
+) nogil:
+    # x is latitude in degree
+    # y is longitude in degree
+    cdef double deg_2_rad = M_PI / 180.0
+    cdef double diff_lat = (x[j] - x[i]) * deg_2_rad
+    cdef double diff_lon = (y[j] - y[i]) * deg_2_rad
+    cdef double arg = (
+        pow(sin(diff_lat/2.0), 2) +
+        cos(x[i]*deg_2_rad) *
+        cos(x[j]*deg_2_rad) *
+        pow(sin(diff_lon/2.0), 2)
+    )
+    # earth radius for WGS84 ellipsoid r ~ 6371km
+    return 12742000.0 * atan2(sqrt(arg), sqrt(1.0-arg))
```

Comment:
I would use the great-circle distance in `deg` as the result, so we don't have to deal with the earth radius. This is also done in Pykrige this way ATM. Then we also don't have to care about `km` vs `m` and leave this decision to the user.

Reply:
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.haversine_distances.html

sklearn takes lat-lon in deg and returns to distance in rad. We could use that as a guide.